service/controller/v14/resource/endpoint/create.go

```diff
@@ -41,34 +42,29 @@ func (r *Resource) newCreateChange(ctx context.Context, obj, currentState, desir
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
-	var ips []string
-	for _, desiredIP := range desiredEndpoint.IPs {
-		if !containsIP(ips, desiredIP) {
-			ips = append(ips, desiredIP)
-		}
-	}
 
 	var createChange *corev1.Endpoints
 	{
-		r.logger.LogCtx(ctx, "level", "debug", "message", "finding out if the endpoint has to be computed")
+		var ips []string
+		for _, ip := range desiredEndpoint.IPs {
+			if !containsIP(ips, ip) {
+				ips = append(ips, ip)
+			}
+		}
 
-		endpointExists := currentEndpoint != nil
-		ipsEmpty := len(ips) == 0
+		endpoint := &Endpoint{
+			IPs:              []string{},
+			ServiceName:      desiredEndpoint.ServiceName,
+			ServiceNamespace: desiredEndpoint.ServiceNamespace,
+		}
 
-		if !endpointExists && !ipsEmpty {
-			r.logger.LogCtx(ctx, "level", "debug", "message", "the endpoint has to be computed")
+		if len(currentEndpoint.IPs) == 0 {
+			endpoint.IPs = ips
+		}
```

Comment:
Re creation being confusing: this check is mind bending. But I understand the reasons. As I said my preference is to do everything in update. However I'm not big on that.

Reply:
That is true. It took me a day to understand how the whole resource works and these conditionals are really not helping. I think we would have them anyway even if we would implement the normal resource interface though.

Reply:
Yes, but then you read top to bottom and understand what is consequence of what. With CRUD I have to jump back and forth and try to remember tiny details. This is my feeling. **Not** saying that implementing basic Resource will 100% fix things.